File: iOS Example/iOS OC Example/NativeViewController.m
//
//  NativeViewController.m
//  iOS OC Example
//
//  Created by 公司 on 2018/6/13.
//

#import "NativeViewController.h"
#import "UIImageView+WebCache.h"

@import AdTiming;

@interface NativeViewController () <ADTNativeDelegate>
@property (nonatomic, strong) NSString *loadId;
@property (nonatomic, strong) ADTNative *native;
@property (nonatomic, strong) ADTNativeAd *nativeAd;
@property (nonatomic, strong) ADTNativeView *nativeView;
@property (nonatomic, strong) UIImageView *iconView;
@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UILabel *bodyLabel;

@end

@implementation NativeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.view addSubview:self.nativeView];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)load:(id)sender {
    self.native = [[ADTNative alloc]initWithPlacmentID:@"109"];
    self.native.delegate = self;
    [self.native loadAd];
}

- (IBAction)show:(id)sender {
    self.nativeView.hidden = NO;
    self.nativeView.nativeAd = self.nativeAd;
    _iconView.image =[UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:self.nativeAd.iconUrl]]];
    _titleLabel.text = self.nativeAd.title;
    _bodyLabel.text = self.nativeAd.body;
}

- (void)ADTNativeDidload:(ADTNativeAd*)nativeAd{
    _nativeAd = nativeAd;
    NSLog(@"广告加载成功");
}

- (ADTNativeView*)nativeView{
    if(!_nativeView){
        _nativeView = [[ADTNativeView alloc]initWithFrame:CGRectMake(0,300, self.view.frame.size.width, 300)];
        _nativeView.mediaView = [[ADTNativeMediaView alloc]initWithFrame:CGRectMake(0, 0, self.view.bounds.size.width, 300)];
        [_nativeView addSubview:_nativeView.mediaView];
        _iconView = [[UIImageView alloc]initWithFrame:CGRectMake(0, 0, 40, 40)];
        [_nativeView addSubview:_iconView];
        
        _titleLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, 270, self.view.frame.size.width, 15)];
        _titleLabel.font = [UIFont systemFontOfSize:13];
        [_nativeView addSubview:_titleLabel];
        
        _bodyLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, 285, self.view.frame.size.width, 15)];
        _bodyLabel.font = [UIFont systemFontOfSize:13];
        [_nativeView addSubview:_bodyLabel];
    }
    return _nativeView;
}

- (void)ADTNativeDidFailToLoad:(NSError*)error{
    NSLog(@"广告加载失败");
}

- (void)ADTNativeWillShow{
    NSLog(@"广告即将展示");
}

- (void)ADTNativeDidClick{
    NSLog(@"点击广告");
}

@end

File: iOS Example/iOS Example/NativeViewController.swift
//
//  NativeViewController.swift
//  iOS Example
//
//  Created by 公司 on 2018/6/13.
//

import UIKit


class NativeViewController: UIViewController {
    
    private lazy var iconView: UIImageView = {
        let icon = UIImageView(frame: CGRect(x: 0, y: 0, width: 40, height: 40))
        return icon
    }()
    
    private lazy var titleLabel: UILabel = {
        let titleLabel = UILabel(frame: CGRect(x: 0, y: 270, width: UIScreen.main.bounds.size.width, height: 15))
        titleLabel.font = UIFont.systemFont(ofSize: 13)
        return titleLabel
    }()
    
    private lazy var bodyLabel: UILabel = {
        let bodyLabel = UILabel(frame: CGRect(x: 0, y: 285, width: UIScreen.main.bounds.size.width, height: 15))
        bodyLabel.font = UIFont.systemFont(ofSize: 13)
        return bodyLabel
    }()
    
    var nativeAd: ADTNativeAd?
    
    private lazy var adtNativeView: ADTNativeView = {
        let view = ADTNativeView(frame: CGRect(x: 0, y: 300, width: self.view.frame.size.width, height: 300))
        view.mediaView = ADTNativeMediaView(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 300))
        view.addSubview(view.mediaView)
        view.addSubview(self.iconView)
        view.addSubview(self.titleLabel)
        view.addSubview(self.bodyLabel)
        
        return view
    }()
    
    private lazy var native: ADTNative = {
        let native = ADTNative(placmentID: "109")
        native.delegate = self
        return native
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.addSubview(self.adtNativeView)
    }
    
    @IBAction func load(_ sender: Any) {
        self.native.loadAd()
    }
    
    @IBAction func show(_ sender: Any) {
        self.adtNativeView.isHidden = false
        self.adtNativeView.nativeAd = self.nativeAd!
        self.iconView.image = UIImage(data: try! Data(contentsOf: URL(string: self.nativeAd?.iconUrl ?? "")!))
        self.titleLabel.text = self.nativeAd?.title
        self.bodyLabel.text = self.nativeAd?.body
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

extension NativeViewController: ADTNativeDelegate {
    func adtNativeDidload(_ nativeAd: ADTNativeAd) {
        self.nativeAd = nativeAd
    }
    
    func adtNativeDidFail(toLoad error: Error) {
        print("nativeAddidFail")
    }
    
    func adtNativeWillShow() {
        print("nativeAdWillExposure")
    }
    
    func adtNativeDidClick() {
        print("nativeAdDidClick")
    }
    
}
